buildtest/menu/build.py

```diff
@@ -74,10 +74,65 @@ def discover_configs(config_file):
         logger.error(msg)
         sys.exit(msg)
 
+    # return all configs with absolute path
+    tmp = [os.path.abspath(config) for config in config_files]
```

Comment:
This is where you'd want to put a more elegant util function to handle returning absolute, symlink resolved paths.

Reply:
Also we should try to avoid a variable name like `tmp` if possible, there is usually a better way to do it.

Reply:
Okay i was able to resolve this and the bug i posted about symlinks by using os.realpath https://docs.python.org/2/library/os.path.html#os.path.realpath which resolves symlinks.

Now this test is working as expected. 
 ```
(buildtest-framework) ssi29@ag-mxg-hulk090> buildtest build -c examples/benchmarks/  -x $PWD/examples/benchmarks
There are no config files to process.
```

My symlink gpfs is now resolving any any subdirectory in symlink works too
```
>>> os.path.realpath('/u/users/ssi29/gpfs')
'/mxg-hpc/users/ssi29'
>>> os.path.realpath('/u/users/ssi29/gpfs/buildtest-framework')
'/mxg-hpc/users/ssi29/buildtest-framework'
```

What won't work is  shell expansion is required then realpath won't expand automatically

```
>>> os.path.realpath('$HOME/gpfs/buildtest-framework')
'/mxg-hpc/users/ssi29/buildtest-framework/$HOME/gpfs/buildtest-framework'
```

But doing shell expansion first then doing realpath would work

```
>>> os.path.realpath(os.path.expandvars('$HOME/gpfs/buildtest-framework'))
'/mxg-hpc/users/ssi29/buildtest-framework'
```

Then it comes to the other odd case of ``~`` if its used when referring to file, so not sure if ``resolve_path`` method should take care of all of these cases in the following order
- Run os.path.expandvars address ``$HOME``
- Run os.path.expandusers address ``~``
- Run os.path.realpath to resolve symlinks
- Finally run ``os.path.exists`` to be absolutely sure its a path.

Reply:
You don't need the tmp here, you can have an iterator that returns to a variable with the same name (not like a dictionary that you can't update in place).
```python
config_files = [os.path.abspath(config) for config in config_files]
```